core/templates/dev/head/services/StateRulesStatsService.js

```diff
@@ -22,58 +22,59 @@ oppia.factory('StateRulesStatsService', [
   function(
       $http, $injector, AngularNameService, AnswerClassificationService,
       ExplorationContextService, UrlInterpolationService) {
-    var STATE_RULES_STATS_URL_TEMPLATE =
-      '/createhandler/state_rules_stats/<exploration_id>/<escaped_state_name>';
-
     return {
       /**
        * TODO(brianrodri): Consider moving this into a visualization domain
        * object.
        *
-       * @param {state} state
+       * @param {Object!} state
        * @return {Boolean} whether given state has an implementation for
        *     displaying the issues overview tab in the State Editor.
        */
       stateSupportsIssuesOverview: function(state) {
         return state.interaction.id === 'TextInput';
       },
+
       /**
        * Returns a promise which will provide details of the given state's
        * answer-statistics.
+       *
+       * @param {Object!} state
        */
-      computeStateRulesStats: function(state, explorationId) {
-        if (explorationId === undefined) {
-          explorationId = ExplorationContextService.getExplorationId();
-        }
-        var stateRulesStatsUrl = UrlInterpolationService.interpolateUrl(
-          STATE_RULES_STATS_URL_TEMPLATE, {
-            exploration_id: explorationId,
-            escaped_state_name: state.name
-          }
-        );
-        return $http.get(stateRulesStatsUrl).then(function(response) {
-          var rulesService = $injector.get(
-            AngularNameService.getNameOfInteractionRulesService(
-              state.interaction.id));
-          var stateRulesStats = {
+      computeStateRulesStats: function(state) {
+        var interactionRulesService = $injector.get(
+          AngularNameService.getNameOfInteractionRulesService(
+            state.interaction.id));
+        var explorationId = ExplorationContextService.getExplorationId();
+
+        return $http.get(
+          UrlInterpolationService.interpolateUrl(
+            '/createhandler/state_rules_stats/<exploration_id>/<state_name>',
+            {exploration_id: explorationId, state_name: state.name})
+        ).then(function(response) {
+          return {
             state_name: state.name,
             exploration_id: explorationId,
-            visualizations_info: response.data.visualizations_info
-          };
+            visualizations_info: response.data.visualizations_info.map(
+              function(vizInfo) {
+                var addressedInfo;
```

Comment:
Might as well just set this to {} initially, then you won't need the "||" in line 75?